### ISSUE TITLE
openai and litellm version confilicts

### ISSUE THREAD

**Issue by zhaochenyang20** [neulab/prompt2model#394]
In the `project.toml`, we do not require a specific version of `openai` and `litellm`. If a user directly runs `pip install .`, then the newest version of `openai` may not support `import openai.error`. However, we use this in our `api_tools.py`. If we choose to use a lower version of `openai`, it is not supported by `litellm`. Thus we need to refactor our `api_tools.py` to support new version of `openai` and `litellm`.

**Comment by viswavi** [comment]
@zhaochenyang20: in `pyproject.toml`, we do specify specific versions of openai and litellm (https://github.com/neulab/prompt2model/blob/main/pyproject.toml)

**Comment by zhaochenyang20** [comment]
Yeah. Make sense to me, but I personally think that we need to keep pace with the latest version of `openai` and `iltellm`. I gonna refactor this later.

**Comment by saum7800** [comment]
I think this has become more important now. I was trying to use base model as `gpt-4-turbo-preview` but was getting litellm errors saying model could not be mapped correctly. I believe this is due to the mismatch. @zhaochenyang20 are you refactoring this?
